src/tests/unit/tests/DetailsView/components/save-assessment-factory.test.tsx

```diff
@@ -1,18 +1,76 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { IMock, Mock, Times } from 'typemoq';
+import { AssessmentDataFormatter } from 'common/assessment-data-formatter';
+import { FileURLProvider } from 'common/file-url-provider';
+import {
+    AssessmentStoreData,
+    AssessmentData,
+} from 'common/types/store-data/assessment-result-data';
 import {
     getSaveButtonForAssessment,
     getSaveButtonForFastPass,
+    SaveAssessmentFactoryDeps,
     SaveAssessmentFactoryProps,
 } from 'DetailsView/components/save-assessment-factory';
 
 describe('SaveAssessmentFactory', () => {
+    let deps: SaveAssessmentFactoryDeps;
     let props: SaveAssessmentFactoryProps;
+    const fileURLProviderMock = Mock.ofType(FileURLProvider);
+    const assessmentDataFormatterMock = Mock.ofType(AssessmentDataFormatter);
+
+    beforeEach(() => {
+        const assessmentStoreData = {
+            assessments: {
+                ['assessment-1']: {
+                    fullAxeResultsMap: null,
+                    generatedAssessmentInstancesMap: null,
+                    manualTestStepResultMap: {
+                        ['assessment-1-step-1']: {
+                            instances: [],
+                            status: 2,
+                            id: 'assessment-1-step-1',
+                        },
+                        ['removed-step']: {
+                            instances: [],
+                            status: 2,
+                            id: '123',
+                        },
+                    },
+                    testStepStatus: {},
+                },
+            } as { [key: string]: AssessmentData },
+        } as AssessmentStoreData;
+
+        deps = {
+            assessmentDataFormatter: assessmentDataFormatterMock.object,
+            fileURLProvider: fileURLProviderMock.object,
+        } as SaveAssessmentFactoryDeps;
+        props = {
+            deps,
+            assessmentStoreData,
+        } as SaveAssessmentFactoryProps;
+    });
 
     describe('getSaveButtonForAssessment', () => {
         test('renders save assessment button', () => {
+            const assessmentData = props.assessmentStoreData.assessments;
+            const formattedAssessmentData = JSON.stringify(assessmentData);
+
+            assessmentDataFormatterMock
+                .setup(a => a.formatAssessmentData(assessmentData))
+                .returns(() => formattedAssessmentData)
+                .verifiable(Times.once());
+
+            fileURLProviderMock
+                .setup(f => f.provideURL([formattedAssessmentData], 'application/json'))
+                .returns(() => 'fileURL');
+
```

Comment:
Is there any particular reason you have `.verifiable(Times.once());` for `assessmentDataFormatterMock` but not `fileURLProviderMock`? Right now, `fileURLProviderMock.verifyAll();` on line 73 doesn't have anything to verify. I could see removing all the verifications or adding one here--whatever you think makes the most sense for the test 🙂

Reply:
Oops, just missed adding that one. Thank you!